infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/fsm/WaitSnapshotApplyState.java

```diff
@@ -65,6 +70,9 @@ public class WaitSnapshotApplyState implements LogReplicationState {
 
     private Optional<Timer.Sample> snapshotSyncApplyTimerSample = Optional.empty();
 
+    @Setter
+    private boolean forcedSnapshotSync;
```

Comment:
This variable is for the optimization of checking the eventTable only when necessary, i.e., only when it is a force snapshot sync. There is no need to check the table otherwise.
This in-memory variable is populated when the event from the eventTable is being processed. And since we only delete the event on a successful completion of snapshot sync (just before going into log_entry sync), its safe to have an in-memory variable indicating a force snapshot sync.

Also, its okay if the node is restarted or even if leader changes between snapshot request and snapshot apply since this variable is only to help optimize reading/deleting the eventTable. 
The event is not deleted until the snapshot sync is successful. So incase the node goes down or the leader changes, the new leader will read the event from the table, populate this variable and trigger a new snapshot sync request.